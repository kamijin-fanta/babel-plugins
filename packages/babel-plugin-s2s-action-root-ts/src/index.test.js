import plugin from './index';
import pluginTester from 'babel-plugin-tester';
import path from 'path';
import fs from 'fs';

pluginTester({
  plugin,
  pluginOptions: {
  },
  tests: [
    {
      title: 'single',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/single/examples/**/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/single/types/action.ts',
      },
      snapshot: true,
    },
    {
      title: 'basic',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/**/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/types/action.ts',
      },
      snapshot: true,
    },
  ],
});
