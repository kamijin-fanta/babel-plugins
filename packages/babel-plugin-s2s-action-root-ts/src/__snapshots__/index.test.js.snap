// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { Action as Container1Action } from \\"../examples/container1/actionTypes.ts\\";
import { Action as Container2Action } from \\"../examples/container2/actionTypes.ts\\";

export type Action = Container1Action | Container2Action;
"
`;

exports[`single 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { Action as Container1Action } from \\"../examples/container1/actionTypes.ts\\";

export type Action = Container1Action;
"
`;
